### PR TITLE
Feat/echam with lco2 emis

### PR DESCRIPTION
**Pull request by chrisdane**
In emission-driven experiments, the atmospheric CO2 concentration is calculated by the model setup and is not prescribed as in (the usual) concentration-driven experiments. For those emission-driven experiments, echam6 needs an additional forcing file: `carbon_emissions.nc` as well as some namelist changes.
 
@pgierz and me made an attempt to implement this: https://github.com/esm-tools/esm_tools/commit/2dcb6675ceb63eb99b3e6f47747787a693bad668

To run such an emission-driven experiment just add
```
with_lco2_emis: True # true for `esm-hist` instead of `historical`; default is False
```
to the `echam` section of the runscript.

The changes in `jsbach.yaml` are just a change in order due to my ocd -_-

Cheers,
Chris

### CODE REVIEW

**Comment by pgierz** [comment]
Thanks Chris! Either myself or Miguel will look at this to put it into the main branch.

We had some nested choose things which were still acting up, but I cannot recall exactly what was the problem.

File: configs/components/jsbach/jsbach.yaml
Comment (pgierz) [inline configs/components/jsbach/jsbach.yaml:187]: This is all the same? Just moved up, or am I missing something?

File: configs/components/echam/echam.yaml
Comment (pgierz) [inline configs/components/echam/echam.yaml:553]: Not really important here, but I would really like to have some link to the various echam options. +1 to Chris for a good comment!

File: configs/components/echam/echam.datasets.yaml
Comment (pgierz) [inline configs/components/echam/echam.datasets.yaml]: Mini correction for formatting:
```suggestion
        # carbon emissions for echam6: with_lco2_emis
```

Still missing, which I still need to set up is a check to ensure that the DKRZ and AWI pool are identical. I did this once by hand, but automating it would be useful.

**Review by pgierz — APPROVED** [review]
Looks good from my side. @mandresm had some thoughts to reduce the repeated statements.

**Comment by mandresm** [comment]
I need the changes from `feat/ssp_generalization` branch which cannot be merged due to some silly blockers that should be solved this week. @chrisdane, can I merge `feat/ssp_generalization` here?

File: configs/components/jsbach/jsbach.yaml
Comment (chrisdane) [reply configs/components/jsbach/jsbach.yaml:187]: yes just order change

File: configs/components/jsbach/jsbach.yaml
Comment (mandresm) [inline configs/components/jsbach/jsbach.yaml]:
```suggestion

```

Comment (mandresm) [reply configs/components/jsbach/jsbach.yaml]: Talking about ocd... I have vim configure to show trailing spaces in red, so it kind of hurts on the eyes

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml:582]: This needs to go in `echam.dataset.yaml`. I'll take care of that.

File: configs/components/echam/echam.yaml
Comment (mandresm) [inline configs/components/echam/echam.yaml:575]: All this repetition can be removed once `feat/ssp_generalization` is merged here.

File: configs/components/echam/echam.datasets.yaml
Comment (mandresm) [inline configs/components/echam/echam.datasets.yaml:148]: We can avoid this repetition when `feat/ssp_generalization` is merged. I'll take care of that fix.

**Review by mandresm — COMMENTED** [review]
This is blocked by #840

**Comment by chrisdane** [comment]
Hi @mandresm 

Yes from my view `feat/ssp_generalization` can be merged, I made one comment there.

File: configs/components/echam/echam.yaml
Comment (chrisdane) [reply configs/components/echam/echam.yaml:553]: Why not important?

**Comment by mandresm** [comment]
Hi Chris, people seems to be discussing a lot on `ssp_generalization`. I won't merge that branch into here until the discussion is finished. Sorry about the inconvenience.

File: configs/components/echam/echam.yaml
Comment (seb-wahl) [reply configs/components/echam/echam.yaml:553]: those options are all described in the ECHAM6 user documentation, although I have to add the sometimes the document is missing some of the options that were introduced just before ECHAM6's development was stopped.

**Comment by chrisdane** [comment]
Can you please continue working on this? (#840 is merged).

**Comment by pgierz** [comment]
I cannot this week, sorry. I've been out sick and am slowly getting back to normal. What is still blocking here?

**Comment by nwieters** [comment]
Hi @mandresm , @pgierz , @chrisdane ,
could someone please resolve these conflicts? Otherwise we can not merge it to release.
Thank you.

**Comment by nwieters** [comment]
Hi @chrisdane , I resolved the conflicts. Can you please review again before merging this PR just to make sure I resolved it correctly? Thank you.

**Comment by chrisdane** [comment]
Yes still looks good to me, thanks a lot @nwieters!

**Comment by pgierz** [comment]
Ok, then I'm pulling the trigger on this. One less one to be sitting around 🙂

**Comment by pgierz** [comment]
#bump #approve-changes #approve_changes (I can never remember if it's dash or underscore)

**Comment by mandresm** [comment]
Hey @pgierz, please bookmark this link https://esm-tools.readthedocs.io/en/latest/esm_tests.html#how-to-approve-changes-on-a-github-pull-request. esm_tests does not handle well bump and approve-changes triggered simulataneously (because it would be really complicated to do it on GitHub Actions). That guide I wrote is a step-by-step guide to navigate the somewhat more complicated workflow (it's not that complicated if you consider that I was doing all this testing before manually ;) ).

File: configs/components/echam/echam.datasets.yaml
Comment (mandresm) [reply configs/components/echam/echam.datasets.yaml:148]: This comment was ignored. Because we got no reply from the user, this PR was never merged.

File: configs/components/echam/echam.yaml
Comment (mandresm) [reply configs/components/echam/echam.yaml:575]: Same thing, we added a lot of repetition. After the rework from Sebastian, many of the repetition in ECHAM where gone. Now they are slowly creeping in again.

**Pull request merged**
